apps/microtvm/zephyr/template_project/microtvm_api_server.py

```diff
@@ -545,6 +546,10 @@ def _find_openocd_serial_port(cls, options):
 
         return ports[0].device
 
+    @classmethod
+    def _find_jlink_serial_port(cls, options):
+        return cls._find_openocd_serial_port(options)
```

Comment:
@mehrdadh I think that ideally `_find_jlink_serial_port()` should not be made merely a wrapper for `_find_openocd_serial_port()`. I understand that the code for the later works well for finding a serial for Jlink too, but openocd is not really involved on detecting the serial port in the end. It looks in Zephyr's CMake configs and goes for a detection using `usb` module. Hence I think that `_find_openocd_serial_port()` should be made a generic code (and rename to something like `_find_serial_port()`) and then be used for both "openocd" and "jlink" flash runners.

`_find_openocd_serial_port()` even contains debug message which cites openocd which is confusing because under the hood for the board in question `JLinkExe` will be used to flash the fw, not openocd.

The `ProjectOption`  `openocd_serial` with also contains "openocd" also becomes incorrect since that option also applies now to `jlink` flash runner, so ideally it should be adapted too I think.

Reply:
Thanks for extra investigation on this. I agree that this is a bit confusing. We could do a refactor here in a follow on PR.